### PR TITLE
Timeseries and Indexed Chart with types

### PR DESCRIPTION
**Pull request by sandikbarr**
As I was looking to clean up and document the try this/try that approach I took with building out chart components, I saw that they could easily be consolidated into two kinds of charts based on their models, so that's what I did. Then I added the documentation we were looking for with #7 and #10 (and #8 which I missed in planning and will talked to Peter about later.) Oh and I know TypeScript a little bit better at this point, so the models are interfaces instead of classes now. Then I updated Publish to use the refactored timeseries chart with type="line". It was about then that I saw we had something strange going on with the time picker, and I think I understand Kerry's comment about "8pm EDT (cuz it's daylight saving time right now?)" because we had both a set of standard and a set of daylight times in the timepicker. Doh. I guess we had a bug on this, probably should have waited for it to be prioritized https://github.com/PRX/publish.prx.org/issues/435

### CODE REVIEW

File: src/lib/src/charts/stacked-bar-category-rotated-chart/stacked-bar-category-rotated-chart.component.ts [hunk 1]
```diff
@@ -1,63 +0,0 @@
-import { Component, Input, OnChanges, Inject, ElementRef } from '@angular/core';
```
Comment (sandikbarr) [inline src/lib/src/charts/stacked-bar-category-rotated-chart/stacked-bar-category-rotated-chart.component.ts]: I dropped support for stacked rotated bar charts. They came up once early on with playback data but didn't go over well, so the component hasn't been kept up to date anyway. We can still bring them back if we decide we want them later. The hard part was figuring out the C3 chart configuration.

File: src/lib/src/datepicker/timepicker.component.ts [hunk 2]
```diff
@@ -18,16 +18,16 @@ export class TimepickerComponent implements OnChanges {
...
-    if (this.date) {
-      this.localTimezone = this.date.toString().match(/(\([A-Za-z\s].*\))/)[1];
-    } else {
-      this.localTimezone = dayGen.toString().match(/(\([A-Za-z\s].*\))/)[1];
+    let match = this.date ? this.date.toString().match(/(\([A-Za-z\s].*\))/) : dayGen.toString().match(/(\([A-Za-z\s].*\))/);
+    if (match && match.length > 0) {
+      this.localTimezone = match[1];
     }
     let day = dayGen.getDate();
+    this.options = [];
```
Comment (sandikbarr) [inline src/lib/src/datepicker/timepicker.component.ts]: The options array was just being pushed onto and not reset every time the date changed.

File: src/lib/src/charts/models/indexed-chart.model.ts [hunk 3]
```diff
@@ -1,7 +1,5 @@
-export class IndexedChartModel {
-  constructor(
-    public data: number[],
-    public label: string,
-    public color: string,
-  ) {  }
+export interface IndexedChartModel {
```
Comment (cavis) [inline src/lib/src/charts/models/indexed-chart.model.ts]: Fancy!

File: src/lib/src/charts/stacked-bar-category-rotated-chart/stacked-bar-category-rotated-chart.component.ts (hunk 1 above)
Comment (cavis) [reply src/lib/src/charts/stacked-bar-category-rotated-chart/stacked-bar-category-rotated-chart.component.ts]: Sounds good - git history ftw!

File: src/lib/src/datepicker/timepicker.component.ts (hunk 2 above)
Comment (cavis) [reply src/lib/src/datepicker/timepicker.component.ts]: good catch!

**Pull request merged**